src/Modules/AuthModule/components/Login/Login.tsx

```diff
@@ -1,144 +1,142 @@
-/* eslint-disable @typescript-eslint/no-unused-vars */
 import { useForm } from "react-hook-form";
-import logo from '../../../../assets/images/PMS 3.png';
+import logo from "../../../../assets/images/PMS 3.png";
 import { Link, useNavigate } from "react-router-dom";
-import {  useState } from "react";
+import { useState } from "react";
 import axios from "axios";
 import { FormData } from "../../../../interfaces/Auth";
 import { useUser } from "../../../../Context/AuthContext";
 import { useToast } from "../../../../Context/ToastContext";
 export default function Login() {
-  // All states here on the top
-  const [showPassword, setShowPassword] = useState<boolean>(false);
-  const { showSuccessToast, showErrorToast } = useToast();
+	// All states here on the top
+	const [showPassword, setShowPassword] = useState<boolean>(false);
+	const { showSuccessToast, showErrorToast } = useToast();
 
-  const {saveAdminData , adminData} = useUser();
-  console.log(adminData)
-  const [spinner, setSpinner] = useState<boolean>(false);
-  const {
-    register,
-    handleSubmit,
-    formState: { errors },
-  } = useForm<FormData>();
-  const navigate = useNavigate();
-  // test done
-  // Function for password visibility
-  //  void indicates that a function does not return any value
-  const togglePasswordVisibility = (): void => {
-    setShowPassword((prevState: boolean) => !prevState);
-  };
+	const { saveAdminData, adminData } = useUser();
+	console.log(adminData);
+	const [spinner, setSpinner] = useState<boolean>(false);
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+	} = useForm<FormData>();
+	const navigate = useNavigate();
+	// test done
+	// Function for password visibility
+	//  void indicates that a function does not return any value
+	const togglePasswordVisibility = (): void => {
+		setShowPassword((prevState: boolean) => !prevState);
+	};
 
-  // senD Data to Api
-  const onSubmit = async (data: FormData) => {
-    setSpinner(true);
+	// senD Data to Api
+	const onSubmit = async (data: FormData) => {
+		setSpinner(true);
 
-    try {
-      const response = await axios.post(
-        "https://upskilling-egypt.com:3003/api/v1/Users/Login",
-        data
-      );
-      localStorage.setItem("adminToken", response?.data?.token);
-      showSuccessToast("Login successfully");
-      localStorage.setItem("token", response.data.token);
-      navigate("/dashboard");
-      saveAdminData && saveAdminData(); // Call saveAdminData only if it exists
-    } catch (error) {
-      if (axios.isAxiosError(error) && error.response) {
-        showErrorToast(error.response.data.message)}
-        else {
-          // Handle other types of errors here
-          showErrorToast("An error occurred.");
-        }
-    } finally {
-      setSpinner(false);
-    }
-  };
+		try {
+			const response = await axios.post(
+				"https://upskilling-egypt.com:3003/api/v1/Users/Login",
```

Comment:
Having the URL hardcoded like this is not a good practice, as this means that if we have the URL in 100 component and the URL has changed that we need to manually update all URLS.

An alternative approach to use axios.create to setup an axios instance that have the base URL and any other configuration, and when using this instance, we will pass only the relative path to the base URL.

We can also store the URL's in a constants file like this one:
https://github.com/AnasKhaled7/PMS-React/blob/main/src/lib/APIs.ts